### DISCUSSION TITLE
mypy error: Call to untyped function "JSONB" in typed context after upgrading to SQLAlchemy 2.0.0

### DISCUSSION THREAD

**Discussion by ijohn** [sqlalchemy/sqlalchemy#9203] [Usage Questions]
Sorry if this has been asked before. I have tried searching Issues, Discussions, and tried Googling.

Upgrading to 2.0.0, from 1.4.46, causes the mypy error stated in discussion title. Relevant snippet of code:

```
import sqlalchemy as sa
from alembic import op
from sqlalchemy.dialects import postgresql

op.create_table("some_table", sa.Column("data", postgresql.JSONB(astext_type=sa.Text()), nullable=True))
```

Anyone knows how to fix? Thanks a lot!

**Comment by CaselIT** [comment]
Hi,

Do you have any other sqlalchemy stub typing package installed?

**Reply by zzzeek** [reply]
I can reproduce it

**Reply by ijohn** [reply]
Thanks all for looking into this!

**Comment by zzzeek** [comment]
I created #9205.  I'm not really sure why the JSON types have this issue and other types of a similar structure seemingly dont.

**Reply by CaselIT** [reply]
It's this covered by the one regarding missing typing?

**Reply by zzzeek** [reply]
?

**Reply by CaselIT** [reply]
I mean the issue here is that it's not typed, and we have this issue tracking it: https://github.com/sqlalchemy/sqlalchemy/issues/6810

**Reply by zzzeek** [reply]
yup, great

**Reply by CaselIT** [reply]
I'll update that issue and pin it later today
